python/cuml/internals/global_settings.py

```diff
@@ -23,8 +24,17 @@ class _GlobalSettingsData(threading.local):  # pylint: disable=R0903
 
     def __init__(self):
         super().__init__()
+        built_with_cuda = True
+        if built_with_cuda and is_cuda_available():
+            default_device_type = 'gpu'
+            default_memory_type = 'global'
```

Comment:
this is why the term `global` is confusing, particularly when `managed` is around as well for users not used to this terminology